packages/embarkjs/src/utils.js

```diff
@@ -53,27 +57,90 @@ let Utils = {
+      //params.gas = gasEstimated + 1000;
+      return toSend.send(params, function(err, transactionHash) {
+        if (err) {
+          return callback(err);
+        }
+        hash = transactionHash;
+        (hashCb || _hashCb)(hash);
+      })
+        .on('receipt', function(receipt) {
+          if (!isContractDeploy || receipt.contractAddress) {
+            callback(null, receipt);
+          }
+        }).then(function(_contract) {
+          if (!hash) {
+            return; // Somehow we didn't get the receipt yet... Interval will catch it
+          }
+          web3.eth.getTransactionReceipt(hash, callback);
+        }).catch(callback);
+      //});
     });
+
+    if (cb) { promise.then(_cb).catch(_cb); return; }
+    return promise;
   }
 };
 
 export default Utils;
+
+var getOwnPropertyDescriptors = Object.getOwnPropertyDescriptors ||
+    function getOwnPropertyDescriptors(obj) {
+      var keys = Object.keys(obj);
+      var descriptors = {};
+      for (var i = 0; i < keys.length; i++) {
+        descriptors[keys[i]] = Object.getOwnPropertyDescriptor(obj, keys[i]);
+      }
+      return descriptors;
+    };
+
+function callbackifyOnRejected(reason, cb) {
+  // `!reason` guard inspired by bluebird (Ref: https://goo.gl/t5IS6M).
+  // Because `null` is a special error value in callbacks which means "no error
+  // occurred", we error-wrap so the callback consumer can distinguish between
+  // "the promise rejected with null" or "the promise fulfilled with undefined".
+  if (!reason) {
+    var newReason = new Error('Promise was rejected with a falsy value');
+    newReason.reason = reason;
+    reason = newReason;
+  }
+  return cb(reason);
+}
+
+export function callbackify(original) {
+  if (typeof original !== 'function') {
+    throw new TypeError('The "original" argument must be of type Function');
+  }
+
+  // We DO NOT return the promise as it gives the user a false sense that
+  // the promise is actually somehow related to the callback's execution
+  // and that the callback throwing will reject the promise.
+  function callbackified() {
+    var args = [];
+    for (var i = 0; i < arguments.length; i++) {
+      args.push(arguments[i]);
+    }
+
+    var maybeCb = args.pop();
+    if (typeof maybeCb !== 'function') {
+      throw new TypeError('The last argument must be of type Function');
+    }
+    var self = this;
+    var cb = function() {
+      return maybeCb.apply(self, arguments);
+    };
+    // In true node style we process the callback on `nextTick` with all the
+    // implications (stack, `uncaughtException`, `async_hooks`)
+    original.apply(this, args)
+      .then(function(ret) { process.nextTick(cb, null, ret); },
+            function(rej) { process.nextTick(callbackifyOnRejected, rej, cb); });
+  }
+
+  Object.setPrototypeOf(callbackified, Object.getPrototypeOf(original));
+  Object.defineProperties(
+    callbackified,
+    getOwnPropertyDescriptors(original)
+  );
+  return callbackified;
+}
```

Comment:
I'm alittle concerned about us introducing our own `callbackify` and `getOwnPropertyDescriptors` polyfills

Is this really necessary? Can't we get those from third party libraries?

Reply:
As explained in the PR description, webpack auto-shim's node's `util` module. I tried to get webpack to use a recent version of the shim, which includes `callbackify` (the one it ships with doesn't) and eventually gave up. But yes, I agree, it would be better to not inline this code.

Reply:
See: https://github.com/embark-framework/embark/pull/1323#discussion_r255754542.